src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,304 @@
+                            ui.printErrorWrongDateFormat();
+                        }
+
+
+                        storage.writeList();
+                    } else {
+                        ui.printErrorNoTime();
+                    }
+                } else if (words[0].equalsIgnoreCase("find")) {
+                    ArrayList<Task> taskFound = new ArrayList<>();
+                    for (Task task : Duke.commandList) {
+                        if (task.name.contains(words[1])) {
+                            taskFound.add(task);
+                        }
+                    }
+                    System.out.println("Here are the matching tasks in your list:");
+                    for (int i = 0; i < taskFound.size(); i++) {
+                        int a = i + 1;
+                        System.out.print(a + ". " + taskFound.get(i));
+                    }
+
+                } else{
+                    ui.printErrorUnderstanding();
+                }
+
+            }
+            ui.printLine();
+            word = input.nextLine();
+        }
+    }
+
+    public String parse(String word) {
+        Ui ui = new Ui();
+        Storage storage = new Storage();
+        String output = "";
+
+            output += ui.printLine();
+            if (word.equalsIgnoreCase("list")) {
+                output += TaskList.printList();
+            } else {
+                String[] words = word.split(" ");
+
+                if (words[0].equalsIgnoreCase("done") && words[1].matches("\\d+")) {
+                    int doneTarget = Integer.parseInt(words[1]);
+                    if (doneTarget > 0 && doneTarget <= Duke.commandList.size()) {
+                        output += TaskList.doneTask(Duke.commandList.get(doneTarget - 1));
+                        storage.writeList();
+
+                    } else {
+                        output += ui.printErrorNotFound();
+                    }
+                } else if (words[0].equalsIgnoreCase("delete") && words[1].matches("\\d+")) {
+                    int deleteTarget = Integer.parseInt(words[1]);
+                    if (deleteTarget > 0 && deleteTarget <= Duke.commandList.size()) {
+                        output += TaskList.deleteTask(Duke.commandList.get(deleteTarget - 1));
+                        storage.writeList();
+                    } else {
+                        output += ui.printErrorNotFound();
+                    }
+
+                } else if (words[0].equalsIgnoreCase("todo")){
+                    if (word.contains("todo ") && !word.substring(5).isEmpty()) {
+                        String substr = word.substring(5);
+                        ToDo task = new ToDo(substr);
+                        output += TaskList.addTask(task);
+                        storage.writeList();
+                    } else {
+                        output += ui.printErrorNoTaskName();
+                    }
+                } else if (words[0].equalsIgnoreCase("deadline")){
```

Comment:
Code starting to look like an arrowhead, using switch case here would definitely help.